frontend/src/app/components/crm/leads/leads.component.ts

```diff
@@ -149,6 +155,47 @@ export class LeadsComponent implements AfterViewInit {
     }
     return `${this.selection.isSelected(row) ? 'deselect' : 'select'} row ${row.leadId + 1}`;
   }
+  /*
+   * This method listens to the contact leads button
+   */
+  contactLead(){
+    let recepientsString : string = "";
+
+    // when no leads are selected
+    if(this.selection.selected.length == 0){
+      alert("Please select at least one lead.");
+      return;
+    }
+
+    let recepients: Array<string> = [];
+
+    for (let i = 0; i < this.selection.selected.length; i++) {
+      if(this.selection.selected[i].columnData.EMAIL == null ){
+         //do nothing
+      }else{
+        recepients.push(this.selection.selected[i].columnData.EMAIL.toString());
+      }
+    }
```

Comment:
You can use [`Array.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to filter out leads with no emails. Then, you could use [`Array.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) to map leads to their email strings.

Reply:
alright, I just thought that it will be better for error handling but it's fine